File: backend/src/Marketplace.Domain/Entities/Product.cs
using Marketplace.Domain.Enums;

namespace Marketplace.Domain.Entities;

public class Product
{
    public Guid Id { get; set; }
    public Guid SellerId { get; set; }
    public Guid? CategoryId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal BasePrice { get; set; }
    public ProductStatus Status { get; set; } = ProductStatus.Draft;
    public string? ImageUrl { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Seller Seller { get; set; } = null!;
    public Category? Category { get; set; }
    public ICollection<ProductVariant> Variants { get; set; } = [];
    public ICollection<Review> Reviews { get; set; } = [];
}

File: frontend/types/product.ts
export type ProductStatus = 'Draft' | 'Active' | 'Archived';

export interface ProductVariant {
  id: string;
  name: string;
  sku: string;
  priceOverride?: number;
  stockQuantity: number;
}

export interface Product {
  id: string;
  sellerId: string;
  storeName: string;
  title: string;
  description: string;
  basePrice: number;
  status: ProductStatus;
  imageUrl?: string;
  categoryId?: string;
  categoryName?: string;
  createdAt: string;
  variants: ProductVariant[];
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface Category {
  id: string;
  name: string;
  parentId?: string;
  children: Category[];
}
